evennia/commands/default/account.py

```diff
@@ -149,53 +149,20 @@ def func(self):
             self.msg("Usage: charcreate <charname> [= description]")
             return
         key = self.lhs
-        desc = self.rhs
-
-        if _MAX_NR_CHARACTERS is not None:
-            if (
-                not account.is_superuser
-                and not account.check_permstring("Developer")
-                and account.db._playable_characters
-                and len(account.db._playable_characters) >= _MAX_NR_CHARACTERS
-            ):
-                plural = "" if _MAX_NR_CHARACTERS == 1 else "s"
-                self.msg(f"You may only have a maximum of {_MAX_NR_CHARACTERS} character{plural}.")
-                return
-        from evennia.objects.models import ObjectDB
+        description = self.rhs or "This is a character."
 
-        typeclass = settings.BASE_CHARACTER_TYPECLASS
+        new_character, errors = self.account.create_character(key=key, description=description, ip=self.session.address)
 
-        if ObjectDB.objects.filter(db_typeclass_path=typeclass, db_key__iexact=key):
-            # check if this Character already exists. Note that we are only
-            # searching the base character typeclass here, not any child
-            # classes.
-            self.msg(f"|rA character named '|w{key}|r' already exists.|n")
```

Comment:
This has been removed, but I don't see this check put back in the `account.create_character` method. Would this not be good to put in `DefaultCharacter.validate_name`?

Reply:
This specific part is an issue. I feel that checking for duplicates is a responsibility of .validate_name() for the Character class, but Evennia doesn't distinguish between PC and NPC for its default Character class...

Although there might be games which DO allow multiple player characters to have the same name. That sounds like it would be confusing, but should the server put such enforcement in place without an ability to turn it off?

Reply:
That's a good point. But either way, it should not be removed from the command then.